ibr/ibr-sdk/class/Layer.js

```diff
@@ -0,0 +1,93 @@
+/**
+* Swap endianness of 32bit numbers.
+* @param {number} val 32 bit number to be swapped.
+* @return {number} 32bit number in swapped endianness.
+*/
+function swap32(val) {
+  return ((val & 0xFF) << 24) |
+           ((val & 0xFF00) << 8) |
+           ((val >> 8) & 0xFF00) |
+           ((val >> 24) & 0xFF);
+}
+
+/**
+ * Constructor of Layer Class.
+ * @param {Objecy} layerData single layer data from JSON Object decoded using
+ Pbf library from raw ibr binary data.
+ */
+function Layer( layerData ) {
+  this.id = layerData.id;
+  this.data = layerData.data;
+
+  // Decode Indices from data.visualization[].coordinate_indices
+  const coordsIndexList = layerData.coordinate_indices;
+  const coordsRangeBuffer = coordsIndexList.buffer.slice(
+      coordsIndexList.byteOffset,
+      coordsIndexList.byteOffset + coordsIndexList.length);
+  const coordsRange = new Uint32Array(coordsRangeBuffer);
+  for (let i = 0; i < coordsRange.length; i++) {
+    coordsRange[i] = swap32(coordsRange[i]);
+  }
```

Comment:
good start! but this can be refined even more to make it even easier for another developer to access this data as coordinates. maybe consider storing this data as line segments since that will bridge even closer to how the data is rendered